File: Day 2/Python/result.py
with open(".\Day 2\Python\data.txt") as f:
    input = f.read().strip().split("\n")


# Part 1
def calculateScore(opp, me):
    if opp == "A":
        if me == "X":
            return 4
        elif me == "Y":
            return 8
        elif me == "Z":
            return 3
    elif opp == "B":
        if me == "X":
            return 1
        elif me == "Y":
            return 5
        elif me == "Z":
            return 9
    elif opp == "C":
        if me == "X":
            return 7
        elif me == "Y":
            return 2
        elif me == "Z":
            return 6



def calculateScore2(opp, me):
    if opp == "A":
        if me == "X":
            return 3
        elif me == "Y":
            return 4
        elif me == "Z":
            return 8
    elif opp == "B":
        if me == "X":
            return 1
        elif me == "Y":
            return 5
        elif me == "Z":
            return 9
    elif opp == "C":
        if me == "X":
            return 2
        elif me == "Y":
            return 6
        elif me == "Z":
            return 7










ans = 0
ans2 = 0
for match in input:
    opp, me = match.split(" ")
    ans += calculateScore(opp, me)
    ans2 += calculateScore2(opp, me)
print(ans)
print(ans2)

File: Day 2/Javascript/result.js
// a x rock 1
// b y paper 2
// c z scissors 3


const input = require('fs')
    .readFileSync("data.txt",)
    .toString()
    .split(/\r?\n/)


let ans = 0;
let ans2 = 0
//Iterate over the data to split into opp picks and my picks
for (let i = 0; i < input.length; i++) {
    let [opp, me] = input[i].split(" ")
    ans += part1(opp, me)
    ans2 += part2(opp, me)
}

console.log({ans})
console.log({ans2})


function part1(opp, me) {
    if (opp == "A") {
        switch(me) {
            case "X":
                return 4
                break
            case "Y":
                return 8
                break
            case "Z":
                return 3
        }
    }else if (opp == "B") {
        switch(me) {
            case "X":
                return 1
                break
            case "Y":
                return 5
                break
            case "Z":
                return 9
        }
    } else if (opp == "C") {
        switch(me) {
            case "X":
                return 7
                break
            case "Y":
                return 2
                break
            case "Z":
                return 6
        } 
    } else {
        console.log(`invalid input ${opp}, ${me}`)
    }
}


function part2(opp, me) {
    if (opp == "A") {
        switch(me) {
            case "X":
                return 3
                break
            case "Y":
                return 4
                break
            case "Z":
                return 8
        }
    }else if (opp == "B") {
        switch(me) {
            case "X":
                return 1
                break
            case "Y":
                return 5
                break
            case "Z":
                return 9
        }
    } else if (opp == "C") {
        switch(me) {
            case "X":
                return 2
                break
            case "Y":
                return 6
                break
            case "Z":
                return 7
        } 
    } else {
        console.log(`invalid input ${opp}, ${me}`)
    }
}

